File: backend/SBL project/SBL.Data/Models/Domain/Author.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SBL.Data.Models.Domain
{
    public class Author
    {
        [JsonProperty("authorId")]
        public int AuthorId { get; set; }

        [JsonProperty("authorName")]
        public string AuthorName { get; set; }
    }
}

File: UI/skyBookLibrary/src/app/data/models/author.ts

export class Author implements IAuthor {
    public authorId?: number
    public authorName: string;


    static create(author: IAuthor) {
        if(!author) return author
        let a = new Author()
        a.authorId = author.authorId
        a.authorName = author.authorName
        return a
    }
}



export interface IAuthor {
    authorId?: number
    authorName: string
}